scripts/core.js

```diff
@@ -64,24 +66,29 @@ var Substitution = (() => {
       if (string === undefined || string === null) return string;
 
       switch (this.inputType) {
+        case this.INPUT_REG_EXP:
         case this.INPUT_TEXT:
           // necessary according to https://stackoverflow.com/q/1520800
           this.regExp.lastIndex = 0;
-          return string.replace(this.regExp, unescape(this.output));
+
+          return (this.outputType === this.OUTPUT_FUNCTION) ?
+            this.replaceWithFn(this.regExp, string) :
+            string.replace(this.regExp, unescape(this.output));
 
         case this.INPUT_WHOLE_WORDS:
           // necessary according to https://stackoverflow.com/q/1520800
           this.regExp.lastIndex = 0;
           return string.replace(this.regExp, (word, index, string) => {
+            // todo by someone who understands: clarify what the following block does and why it does it
```

Comment:
I don't remember the details because I wrote it many years ago, but it has to do with correct functioning of replace whole words with non-ASCII characters, including respecting the special strings `$$`, `$&`, etc. It was added in df6fe586.